src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ConversationService.java

```diff
@@ -0,0 +1,393 @@
+
+    public ConversationService(ConversationDTOService conversationDTOService, UserRepository userRepository, ChannelRepository channelRepository,
+            ConversationParticipantRepository conversationParticipantRepository, ConversationRepository conversationRepository, SimpMessageSendingOperations messagingTemplate,
+            OneToOneChatRepository oneToOneChatRepository, PostRepository postRepository, GroupChatRepository groupChatRepository) {
+        this.conversationDTOService = conversationDTOService;
+        this.userRepository = userRepository;
+        this.channelRepository = channelRepository;
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.conversationRepository = conversationRepository;
+        this.messagingTemplate = messagingTemplate;
+        this.oneToOneChatRepository = oneToOneChatRepository;
+        this.postRepository = postRepository;
+        this.groupChatRepository = groupChatRepository;
+    }
+
+    /**
+     * Gets the conversation with the given id
+     *
+     * @param conversationId the id of the conversation
+     * @return the conversation with the given id
+     */
+    public Conversation getConversationById(Long conversationId) {
+        return conversationRepository.findByIdElseThrow(conversationId);
+    }
+
+    /**
+     * Checks whether the user is a member of the conversation
+     *
+     * @param conversationId the id of the conversation
+     * @param userId         the id of the user
+     * @return true if the user is a member of the conversation, false otherwise
+     */
+    public boolean isMember(Long conversationId, Long userId) {
+        return conversationParticipantRepository.findConversationParticipantByConversationIdAndUserId(conversationId, userId).isPresent();
+    }
+
+    /**
+     * Gets the conversation in a course for which the user is a member
+     *
+     * @param courseId       the id of the course
+     * @param requestingUser the user for which the conversations are requested
+     * @return the conversation in the course for which the user is a member
+     */
+    public List<ConversationDTO> getConversationsOfUser(Long courseId, User requestingUser) {
+        var oneToOneChatsOfUser = oneToOneChatRepository.findActiveOneToOneChatsOfUserWithParticipantsAndUserGroups(courseId, requestingUser.getId());
+        var channelsOfUser = channelRepository.findChannelsOfUser(courseId, requestingUser.getId());
+        var groupChatsOfUser = groupChatRepository.findGroupChatsOfUserWithParticipantsAndUserGroups(courseId, requestingUser.getId());
+
+        var conversations = new ArrayList<Conversation>();
+        conversations.addAll(oneToOneChatsOfUser);
+        conversations.addAll(channelsOfUser);
+        conversations.addAll(groupChatsOfUser);
+        return conversations.stream().map(conversation -> conversationDTOService.convertToDTO(conversation, requestingUser)).collect(Collectors.toList());
```

Comment:
[Docs](https://docs.artemis.cit.tum.de/dev/guidelines/server/?highlight=tolist#general-best-practices)
```suggestion
        return conversations.stream().map(conversation -> conversationDTOService.convertToDTO(conversation, requestingUser)).toList();
```